src/plugins/intel_gpu/tests/unit/test_cases/memory_test.cpp

```diff
@@ -595,6 +596,58 @@ class memory_pool: public ::testing::Test {
                       std::static_pointer_cast<fully_connected_inst>(network.get_primitive("relu2"))->output_memory_ptr()->buffer_ptr());
         }
     }
+
+    void test_dynamic_mem_reuse_for_null_sel_impl() {
+        auto& engine = get_test_engine();
+
+        const int32_t input_f = 3;
+
+        auto input_static_layout = layout{ ov::PartialShape{ 2, input_f, 1, 2 }, data_types::f32, format::bfyx };
+        auto input_dyn_layout = layout{ ov::PartialShape{ 2, input_f, ov::Dimension(), 2 }, data_types::f32, format::bfyx };
+
+
+        auto input_actual_layout1 = layout{ ov::PartialShape{ 2, input_f, 1, 2 }, data_types::f32, format::bfyx};
+        auto input_actual_layout2 = layout{ ov::PartialShape{ 1, input_f, 1, 2 }, data_types::f32, format::bfyx};
+        auto input_data1 = engine.allocate_memory(input_actual_layout1);
+        auto input_data2 = engine.allocate_memory(input_actual_layout2);
+        auto fc_weights_data = engine.allocate_memory({ ov::PartialShape{ 2, input_f, 1, 1 }, data_types::f32, format::bfyx});
+
+        set_values(input_data1, { -0.5f, 2.0f, 0.5f, -0.5f, 2.0f, 0.5f });
+        set_values(input_data2, { 0.5f, -2.0f, -0.5f, -0.5f, 2.0f, 0.5f,
+                                -0.5f, 2.0f, 0.5f, -0.5f, 2.0f, 0.5f });
```

Comment:
The sizes of data and layouts of `input_data1` and `input_data2` are not matched.
Additionally, this test doesn't check the correctness of the results.
Please update this test to check it.

Reply:
Layout size issue is fixed.
Result value doesn't check because the goal of this test is cheking 'not reuse mem' so it only checks memomy buffer pointer value.